src/ConsoleInputArgument.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console;
+
+use ReflectionParameter;
+
+final class ConsoleInputArgument
```

Comment:
Circling back to my previous comment: I think this class is trying to do too many things. Either it represents the input given by the user via the command in a structured way, or it represents the definition of the expected value.

Reply:
True that - I like the current design much better - thanks for all your comments!